[package] boost/1.72.0: Boost.Log can't be built with boost:filesystem_no_deprecated=True

### Package and Environment Details
  * Package Name/Version: **boost/1.72.0**
  * Operating System+version: **Windows 10**
  * Compiler+version: **VS2017 Update 9 (v15.9.20)**
  * Conan version: **conan 1.24.0**
  * Python version: **Python 3.8.1**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=15
build_type=Release
[options]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)
Build boost without excluding Boost.Log + enable boost:filesystem_no_deprecated=True
conanfile.txt
```
[generators]
cmake

[requires]
boost/1.72.0

[options]
boost:shared=False
boost:header_only=False
boost:error_code_header_only=False
boost:system_no_deprecated=True
boost:asio_no_deprecated=True
boost:filesystem_no_deprecated=True
boost:layout=system
boost:magic_autolink=False
boost:zlib=False
boost:bzip2=False
boost:lzma=False
boost:zstd=False
boost:without_math=True
boost:without_wave=True
boost:without_container=True
boost:without_contract=True
boost:without_exception=True
boost:without_graph=True
boost:without_iostreams=True
boost:without_locale=True
boost:without_log=False
boost:without_program_options=True
boost:without_random=True
boost:without_regex=True
boost:without_mpi=True
boost:without_serialization=True
boost:without_coroutine=True
boost:without_fiber=True
boost:without_context=True
boost:without_timer=True
boost:without_thread=False
boost:without_chrono=False
boost:without_date_time=True
boost:without_atomic=False
boost:without_filesystem=True
boost:without_system=True
boost:without_graph_parallel=True
boost:without_python=True
boost:without_stacktrace=True
boost:without_test=True
boost:without_type_erasure=True

[imports]
```
conan install . --build missing


### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
 cl /Zm800 -nologo @"C:\.conan\727b56\1\boost\bin.v2\libs\log\build\msvc-14.1\rls\adrs-mdl-64\lnk-sttc\thrdp-wn32\thrd-mlt\text_file_backend.obj.rsp" 



text_file_backend.cpp

libs\log\src\text_file_backend.cpp(77): error C2039: 'filesystem_error': is not a member of 'boost::filesystem'

.\boost/filesystem/convenience.hpp(29): note: see declaration of 'boost::filesystem'

libs\log\src\text_file_backend.cpp(77): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int

libs\log\src\text_file_backend.cpp(77): error C2146: syntax error: missing ';' before identifier 'filesystem_error'

libs\log\src\text_file_backend.cpp(77): warning C4091: '': ignored on left of 'int' when no variable is declared

libs\log\src\text_file_backend.cpp(800): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'boost::log::v2s_mt_nt6::sinks::`anonymous-namespace'::filesystem_error'

libs\log\src\text_file_backend.cpp(803): note: The initializer contains too many elements

libs\log\src\text_file_backend.cpp(800): error C2672: 'boost::exception_detail::throw_exception_': no matching overloaded function found

libs\log\src\text_file_backend.cpp(919): error C2039: 'directory_iterator': is not a member of 'boost::filesystem'

.\boost/filesystem/convenience.hpp(29): note: see declaration of 'boost::filesystem'

libs\log\src\text_file_backend.cpp(919): error C2065: 'directory_iterator': undeclared identifier

libs\log\src\text_file_backend.cpp(919): error C2146: syntax error: missing ';' before identifier 'it'

libs\log\src\text_file_backend.cpp(919): error C3861: 'it': identifier not found

libs\log\src\text_file_backend.cpp(919): error C2563: mismatch in formal parameter list

libs\log\src\text_file_backend.cpp(919): error C2568: ',': unable to resolve function overload

libs\log\src\text_file_backend.cpp(919): note: could be 'range_iterator<const T,void>::type boost::range_adl_barrier::end(const T &)'

libs\log\src\text_file_backend.cpp(919): note: or       'range_iterator<C,void>::type boost::range_adl_barrier::end(T &)'

libs\log\src\text_file_backend.cpp(921): error C2065: 'it': undeclared identifier

libs\log\src\text_file_backend.cpp(921): error C2563: mismatch in formal parameter list

libs\log\src\text_file_backend.cpp(921): error C2568: '!=': unable to resolve function overload

libs\log\src\text_file_backend.cpp(921): note: could be 'range_iterator<const T,void>::type boost::range_adl_barrier::end(const T &)'

libs\log\src\text_file_backend.cpp(921): note: or       'range_iterator<C,void>::type boost::range_adl_barrier::end(T &)'

libs\log\src\text_file_backend.cpp(923): error C2065: 'it': undeclared identifier

libs\log\src\text_file_backend.cpp(925): error C2027: use of undefined type 'boost::filesystem::directory_entry'

.\boost/filesystem/path_traits.hpp(48): note: see declaration of 'boost::filesystem::directory_entry'

libs\log\src\text_file_backend.cpp(926): error C2027: use of undefined type 'boost::filesystem::directory_entry'

.\boost/filesystem/path_traits.hpp(48): note: see declaration of 'boost::filesystem::directory_entry'

libs\log\src\text_file_backend.cpp(1389): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'boost::log::v2s_mt_nt6::sinks::`anonymous-namespace'::filesystem_error'

libs\log\src\text_file_backend.cpp(1392): note: The initializer contains too many elements

libs\log\src\text_file_backend.cpp(1389): error C2672: 'boost::exception_detail::throw_exception_': no matching overloaded function found

...failed compile-c-c++ C:\.conan\727b56\1\boost\bin.v2\libs\log\build\msvc-14.1\rls\adrs-mdl-64\lnk-sttc\thrdp-wn32\thrd-mlt\text_file_backend.obj...
```
Full log if needed: https://gist.github.com/Nekto89/ff25c5c0b41182c12f0b060053b3d29c

</details>

--- a/recipes/boost/all/conandata.yml
+++ b/recipes/boost/all/conandata.yml
@@ -67,3 +67,5 @@ patches:
       base_path: "boost_1_72_0"
     - patch_file: "patches/0001-revert-cease-dependence-on-range.patch"
       base_path: "boost_1_72_0"
+    - patch_file: "patches/boost_log_filesystem_no_deprecated_1_72.patch"
+      base_path: "boost_1_72_0"

--- a/recipes/boost/all/patches/boost_log_filesystem_no_deprecated_1_72.patch
+++ b/recipes/boost/all/patches/boost_log_filesystem_no_deprecated_1_72.patch
@@ -1,0 +1,13 @@
+diff --git a/libs/log/src/text_file_backend.cpp b/libs/log/src/text_file_backend.cpp
+index 50cef3b..d5fd02e 100644
+--- a/libs/log/src/text_file_backend.cpp
++++ b/libs/log/src/text_file_backend.cpp
+@@ -39,6 +39,8 @@
+ #include <boost/type_traits/is_same.hpp>
+ #include <boost/system/error_code.hpp>
+ #include <boost/system/system_error.hpp>
++#include <boost/filesystem/directory.hpp>
++#include <boost/filesystem/exception.hpp>
+ #include <boost/filesystem/path.hpp>
+ #include <boost/filesystem/fstream.hpp>
+ #include <boost/filesystem/operations.hpp>

